contracts/RewardController.vy

```diff
@@ -460,37 +480,54 @@ def update_many(dat: Bytes[MAX_PAYLOAD_SIZE]) -> DynArray[EnhancedReward, MAX_PA
         deviation = self._calc_deviation(scid, convert(raw_deviation, uint256))
         time_since = convert(rec.new_timestamp - rec.old_timestamp, uint256) * EIGHTEEN_DECIMAL_NUMBER_U
 
-        # calculate reward
-        time_reward, deviation_reward = self._calc_reward(convert(time_since, int256)//1000,
-                                                          convert(deviation, int256),
-                                                          coeff)
-     
-        # calculate reward multiplier
-        reward_mult = self._calc_reward_mult(scid, time_since//1000)
-
-        # adjust rewards with multiplier
-        time_reward_adj = reward_mult * time_reward // EIGHTEEN_DECIMAL_NUMBER
-        deviation_reward_adj = reward_mult * deviation_reward // EIGHTEEN_DECIMAL_NUMBER
-
-        time_reward_adj_u = convert(time_reward_adj, uint256)
-        deviation_reward_adj_u = convert(deviation_reward_adj, uint256)
-
-        # store rewards
-        self.rewards[msg.sender] += time_reward_adj_u + deviation_reward_adj_u
-        self.total_rewards += time_reward_adj_u + deviation_reward_adj_u
-
-        log OracleUpdated(updater=msg.sender, system_id=sid, chain_id=cid,
-                          new_value=new_gasprice, raw_deviation=raw_deviation,
-                          time_since=convert(time_since//10**21, uint48),
-                          time_reward=time_reward_adj_u, deviation_reward=deviation_reward_adj_u,
-                          reward_mult=reward_mult)
-
-        rewards.append(EnhancedReward(system_id=sid,
-                                      chain_id=cid,
-                                      height=rec.old_height,
-                                      gas_price=old_gasprice,
-                                      time_reward=time_reward_adj_u,
-                                      deviation_reward=deviation_reward_adj_u))
+        # Dont reward
+        if not self.rewards_enabled:
+            log OracleUpdated(updater=msg.sender, system_id=sid, chain_id=cid,
+                              new_value=new_gasprice, raw_deviation=raw_deviation,
+                              time_since=convert(time_since//10**21, uint48),
+                              time_reward=0, deviation_reward=0,
+                              reward_mult=0)
+
+            rewards.append(EnhancedReward(system_id=sid,
```

Comment:
You grant re

Reply:
frozen mean revert and don't allow updates
rewards_disabled means update but give 0 reward. 0 rewards is only possible if rewards_disabled

Reply:
What's the scenario for wanting this to be disabled but not frozen?

Reply:
One scenario is deploy w/ rewards off, so we can test to make sure updates are working, but don't emit rewards yet.
Another one could be to pause rewards while we fix something.